audio-analysis/sqlConnection.py

```diff
@@ -0,0 +1,52 @@
+import pymysql
+from sqlalchemy import create_engine, MetaData, Table, Column, Integer, String, Float, Numeric
+
+#### SQL CONNECTION INFO ####
+# Currently working on localhost
+# MAKE SURE YOU HAVE A DATABASE CALLED 'climactic_test'
+
+engine = create_engine('mysql+pymysql://root@localhost/climactic_test') # Make sure you use the correct username "root" and password
+
+#SQL Connection Setup - Checks if table exists and if URL to be input already exists
+def sqlConnectionSetup():
+    global engine
+    table_exists = False #Variable to check if table exists
+    # Create a metadata instance
+    meta = MetaData(engine)
+    #meta.reflect(bind=engine)
+
+    # Check if table already exists
+    for _t in meta.tables:
+        if _t == 'test_table':
+            table_exists = True
+
+    # If table doesn't exist, create it
+    if table_exists == False:
+        # Declare a table
+        table = Table('test_table',meta,
+                    Column('id',Integer, primary_key=True),
+                    Column('index',Integer),
+                    Column('end_time_s',Integer),
+                    Column('polarity',Float),
+                    Column('start_time_s',Integer),
+                    Column('subjectivity',Float),
+                    Column('url',String(length=500)),
+                    Column('word',String(length=500)),
+                    Column('amplitude',Float),
+                    Column('amplitude_peak',Integer),
+                )
+        # Create all tables
+        meta.create_all()
+
+#Function checks if url exists in table already
+def urlExists(url):
+    global engine
+    # Check if URL already exists
+    result = engine.execute('SELECT * FROM `test_table` WHERE `url`="' + url + '"')
```

Comment:
If you use `SELECT COUNT(*)` you can get a number back and don't need to use `result.rowcount`

Reply:
Yea tried doing this, unfortunately the `engine.execute` function returns what is called a "ResultProxy" which is like it's own object. 
Even if I output the COUNT I would have to query the "ResultProxy" to get the actual value. Thought it might be faster to just get rowcount - what you think? Could be wrong tbh

Reply:
hmm that's sus. I briefly used sqlalchemy during a coop and found it pretty whack. I'm good with the way it is then. As long as it works lol

Reply:
Some mischievious sqlalchemy smoke and mirrors.

Reply:
Yea kind of weird. Hopefully this doesn't affect runtime but I'll assume we're good. SQLalchemy is a little weird but since we started might as well just keep going.